src/Nethermind/Nethermind.Blockchain/Synchronization/SyncModeSelector.cs

```diff
@@ -38,92 +38,102 @@ public SyncModeSelector(ISyncProgressResolver syncProgressResolver, IEthSyncPeer
             _syncPeerPool = syncPeerPool ?? throw new ArgumentNullException(nameof(syncPeerPool));
             _syncConfig = syncConfig ?? throw new ArgumentNullException(nameof(syncConfig));
             _logger = logManager.GetClassLogger() ?? throw new ArgumentNullException(nameof(logManager));
-            
+
             Current = SyncMode.NotStarted;
         }
 
         public SyncMode Current { get; private set; }
 
         public bool IsParallel => Current == SyncMode.FastBlocks || Current == SyncMode.StateNodes;
-        
+
         public void Update()
         {
             if (_syncPeerPool.PeerCount == 0)
             {
                 return;
             }
 
+            if (_syncConfig.HmbSync)
+            {
+                if (Current != SyncMode.HmbSync)
+                {
+                    ChangeSyncMode(SyncMode.HmbSync);
```

Comment:
We need to switch from BeamSync to FullSync when we download all the needed headers, blocks, receipts and state

Reply:
yeah, but this is not implemented yet - this commit just introduces first beam sync classes but is mainly about refactoring

Reply:
but good point, I did not think of that yet - there re some following BeamSync commits that are not included in the PR

Reply:
sure, just pointing out.